src/components/PasswordStrengthIndicator.js

```diff
@@ -0,0 +1,39 @@
+import React from "react";
+
+import "./PasswordStrengthIndicator.css";
+
+function getColor(strength) {
+  if (strength < 3) {
+    return "red";
+  }
+  if (strength < 5) {
+    return "orange";
+  }
+  if (strength < 7) {
+    return "yellow";
+  }
+  return "green";
+}
+
+function StrengthIndicator(props) {
+  const { strength, description } = props.determineStrength(props.password);
+  if (strength < 0 || strength > 10) {
+    throw new Error("strength is out of bounds 0>=strength>=10");
+  }
+  const color = getColor(strength);
+  return (
+    <div className="password-strength">
+      <h3>{props.name}</h3>
+      <svg
+        viewBox="0 0 300 10"
+        alt={`Password strength rating: ${strength}/10`}
+      >
+        <rect fill="#ddd" x="0" y="0" width="300" height="10" />
+        <rect fill={color} y="0" width={strength * 30} height="10"></rect>
+      </svg>
+      <p>{description}</p>
+    </div>
+  );
+}
```

Comment:
this is my suggestion for how to standardize password strength indicators into one consistent look and feel. Basically, for each one we derive a function that takes the password as an input and returns a strength indicator between 0 and 10 inclusive, and a short description of the strength of the password. Might take some adjusting since I know not all ratings use a 10 point scale. Thoughts on this approach?

Reply:
It's definitely an improvement over the "weak" to "strong" scale, and gives some actual feedback, so I approve

Reply:
Cool! And if anything does come up it should be easy to adjust later.

Reply:
I agree this is a good way to implement it. Would it be shown on the page too, or will the meter just be handled in the backend?

Reply:
Yeah the frontend will show the text of the description and a colored bar that has a length based on the strength value (full bar = 10/10) for each strength indicator we create